emu/src/cpu/arm7tdmi.rs

```diff
@@ -228,6 +228,14 @@ impl Arm7tdmi {
         }
     }
 
+    fn uncond_branch(&mut self, op_code: ThumbModeOpcode) -> Option<u32> {
+        let offset = (op_code.get_bits(0..=10) << 1) as i32;
```

Comment:
We need to sign-extend it. If we just do `as i32` it doesn't work.

Example in `u8/i8` where we want to sign-extend a 3-bits number:

```rust
let a: u8 = 0b110; // To sign-extend
let b: i8 = a as i8;
```

In this case `b == 0b0000_0110` while we would it to be `b == 0b1111_1110`.

To see how to sign-extend you can check the function `branch` for ARM. And while you're there maybe you can think how to move this logic somewhere so we don't copy paste it every time :P

Reply:
I also refactored :)

Reply:
You moved the logic somewhere else? I don't see it :(

Reply:
Yes is a function `sign_extend`... Hit refresh :D